solc/CommandLineInterface.cpp

```diff
@@ -411,16 +421,33 @@ bool CommandLineInterface::readInputFilesAndConfigureFileReader()
 		}
 
 		// NOTE: we ignore the FileNotFound exception as we manually check above
-		m_fileReader.setSource(infile, readFileAsString(infile.string()));
-		m_fileReader.allowDirectory(boost::filesystem::path(boost::filesystem::canonical(infile).string()).remove_filename());
+		string fileContent = readFileAsString(infile.string());
+		if (m_options.input.mode == InputMode::StandardJson)
+		{
+			solAssert(!m_standardJsonInput.has_value(), "");
+			m_standardJsonInput = move(fileContent);
+		}
+		else
+		{
+			m_fileReader.setSource(infile, move(fileContent));
+			m_fileReader.allowDirectory(boost::filesystem::canonical(infile).remove_filename());
+		}
 	}
 
 	if (m_options.input.addStdin)
-		m_fileReader.setSource(g_stdinFileName, readUntilEnd(m_sin));
+	{
+		if (m_options.input.mode == InputMode::StandardJson)
+		{
+			solAssert(!m_standardJsonInput.has_value(), "");
+			m_standardJsonInput = readUntilEnd(m_sin);
+		}
+		else
+			m_fileReader.setSource(g_stdinFileName, readUntilEnd(m_sin));
+	}
 
-	if (m_fileReader.sourceCodes().size() == 0)
+	if (m_fileReader.sourceCodes().empty() && !m_standardJsonInput.has_value())
 	{
-		serr() << "No input files given. If you wish to use the standard input please specify \"-\" explicitly." << endl;
+		serr() << "All specified input files either do not exist or are not regular files." << endl;
```

Comment:
Mostly because it's the prevalent style used in this file and I don't think the difference matters for us here. As for readability, I slightly prefer `endl` because `\n` looks like noise to me :) but I don't feel too strongly about either.

If we want to change this, we should go over the whole file (and maybe more files) so I'd rather not do it in this PR.
It might actually be a `good first issue` for someone so maybe create an issue for it?

Reply:
Oh, I just noticed that I used `\n` in the base path message. Changed that to `endl` now for consistency.